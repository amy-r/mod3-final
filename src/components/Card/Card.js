import React, { Component } from 'react';
import { getMembers } from '../../helper/helper';
import './Card.css';
import PropTypes from 'prop-types';

class Card extends Component {
  constructor() {
    super();
    this.state = {
      swornMembers: [],
      names: []
    }
  }
    
  handleClick = async () => {
    const members = await this.state.swornMembers.map( async (url)  => {
      const response = await getMembers(url);
      return response.name
    })
    const names = await Promise.all(members)
    this.setState({ names })
    console.log(this.state.names)
  }

  componentDidMount = () => {
    this.setState({swornMembers: this.props.swornMembers})
  }

  showMembers = () => {
    if (this.state.names.length) {
      return this.state.names.map( (name) => {
        return <p className='name'> {name}, </p>
      })    
    }
  }

  render() {
    const {name, founded, seats, titles, coatOfArms, ancestralWeapons, words, swornMembers} = this.props
    return (
      <div className ='Card' onClick ={this.handleClick}>
        <h2> {name} </h2>
        <h3> Founded: {founded} </h3>
        <p> Seats: {seats} </p>
        <p> Titles: {titles} </p>
        <p> Coat Of Arms: {coatOfArms} </p>
        <p> Ancestral Weapons: {ancestralWeapons} </p> 
        <p> Words: {words} </p>
        <p> Sworn Members:</p> 
        <div className = 'members'>
          {this.showMembers()}
        </div>  
      </div>
    )
  }
}

Card.propTypes = {
  name: PropTypes.string.isRequired,
  founded: PropTypes.string.isRequired,
  seats: PropTypes.arrayOf(PropTypes.string).isRequired,
  titles: PropTypes.arrayOf(PropTypes.string).isRequired,
  coatOfArms: PropTypes.string.isRequired,
  ancestralWeapons: PropTypes.arrayOf(PropTypes.string).isRequired,
  words: PropTypes.string.isRequired,
  swornMembers: PropTypes.arrayOf(PropTypes.string).isRequired
};
export default Card;