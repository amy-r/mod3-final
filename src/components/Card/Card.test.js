import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('should match the snapshot', () => {
    const house = {
      name: 'fake',
      founded: 'fake',
      seats: ['fake'],
      titles: ['fake'],
      coatOfArms: 'fake',
      ancestralWeapons: ['fake'],
      words: 'fake',
      swornMembers: ['fake']
    }
    const wrapper = shallow(<Card {...house}/>)

    expect(wrapper).toMatchSnapshot();
  })
})