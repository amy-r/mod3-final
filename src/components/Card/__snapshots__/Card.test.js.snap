// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    ancestralWeapons={
        Array [
            "fake",
          ]
    }
    coatOfArms="fake"
    founded="fake"
    name="fake"
    seats={
        Array [
            "fake",
          ]
    }
    swornMembers={
        Array [
            "fake",
          ]
    }
    titles={
        Array [
            "fake",
          ]
    }
    words="fake"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
           
          fake
           
</h2>,
        <h3>
           Founded: 
          fake
           
</h3>,
        <p>
           Seats: 
          fake
           
</p>,
        <p>
           Titles: 
          fake
           
</p>,
        <p>
           Coat Of Arms: 
          fake
           
</p>,
        <p>
           Ancestral Weapons: 
          fake
           
</p>,
        <p>
           Words: 
          fake
           
</p>,
        <p>
           Sworn Members:
</p>,
        <div
          className="members"
/>,
      ],
      "className": "Card",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "fake",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "fake",
          " ",
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Founded: ",
            "fake",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Founded: ",
          "fake",
          " ",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Seats: ",
            Array [
              "fake",
            ],
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Seats: ",
          "fake",
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Titles: ",
            Array [
              "fake",
            ],
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Titles: ",
          "fake",
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Coat Of Arms: ",
            "fake",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Coat Of Arms: ",
          "fake",
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Ancestral Weapons: ",
            Array [
              "fake",
            ],
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Ancestral Weapons: ",
          "fake",
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Words: ",
            "fake",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " Words: ",
          "fake",
          " ",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Sworn Members:",
        },
        "ref": null,
        "rendered": " Sworn Members:",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "members",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
             
            fake
             
</h2>,
          <h3>
             Founded: 
            fake
             
</h3>,
          <p>
             Seats: 
            fake
             
</p>,
          <p>
             Titles: 
            fake
             
</p>,
          <p>
             Coat Of Arms: 
            fake
             
</p>,
          <p>
             Ancestral Weapons: 
            fake
             
</p>,
          <p>
             Words: 
            fake
             
</p>,
          <p>
             Sworn Members:
</p>,
          <div
            className="members"
/>,
        ],
        "className": "Card",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "fake",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "fake",
            " ",
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Founded: ",
              "fake",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Founded: ",
            "fake",
            " ",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Seats: ",
              Array [
                "fake",
              ],
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Seats: ",
            "fake",
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Titles: ",
              Array [
                "fake",
              ],
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Titles: ",
            "fake",
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Coat Of Arms: ",
              "fake",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Coat Of Arms: ",
            "fake",
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Ancestral Weapons: ",
              Array [
                "fake",
              ],
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Ancestral Weapons: ",
            "fake",
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Words: ",
              "fake",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " Words: ",
            "fake",
            " ",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Sworn Members:",
          },
          "ref": null,
          "rendered": " Sworn Members:",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "members",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
